src/nvim/syntax.c

```diff
@@ -6874,8 +6874,20 @@ highlight_color (
   else if (!(TOLOWER_ASC(what[0]) == 'b' && TOLOWER_ASC(what[1]) == 'g'))
     return NULL;
   if (modec == 'g') {
-    if (fg)
+    if (what[2] == '#') {
```

Comment:
Need to check `ui_rgb_attached()`. 

    if (what[2] == '#' && ui_rgb_attached()) {